### PR TITLE
Upgrade to symfony 7 compatibility

### PR DESCRIPTION
**Pull request by tomglvng**

### CODE REVIEW

**Comment by lasdou** [comment]
Why not going directly php 8.3 as php 8.2 [active support will end in 3 months  31 Dec 2024](https://www.php.net/supported-versions.php) ?

**Review by lasdou — COMMENTED** [review]
Why not going directly php 8.3 as php 8.2 [active support will end in 3 months 31 Dec 2024](https://www.php.net/supported-versions.php) ?

File: composer.json [hunk 1]
```diff
@@ -20,19 +20,19 @@
...
-    "php": ">=8.0",
+    "php": ">=8.2",
     "psr/http-message": "^1.0",
     "psr/http-client": "^1.0",
     "psr/http-factory": "^1.0",
     "psr/http-message-implementation": "^1.0|^2.0",
     "php-http/httplug": "^2.0",
     "php-http/discovery": "^1.6",
     "php-http/multipart-stream-builder": "^1.0",
-    "symfony/options-resolver": "^5.4|^6.0|^7.0"
```
[the remark this answers is not in the document]
Comment (vjrabanelly) [reply composer.json:31]: Also, how does this bring compatibility with Symfony 7? It seems it was already in the list, so I don't see what it adds.

[the remark this answers is not in the document]
Comment (tomglvng) [reply composer.json:31]: Symfony 7 needs at least PHP 8.2

File: composer.json [hunk 2]
```diff
@@ -20,19 +20,19 @@
     }
   },
   "require": {
-    "php": ">=8.0",
+    "php": ">=8.2",
     "psr/http-message": "^1.0",
```
[the remark this answers is not in the document]
Comment (tomglvng) [reply composer.json:24]: Ok, so I need to upgrade psr/http-message

**Comment by tomglvng** [comment]
> Why not going directly php 8.3 as php 8.2 [active support will end in 3 months 31 Dec 2024](https://www.php.net/supported-versions.php) ?

We should always be compatible with most people without compromising security : the last PHP maintained PHP version is 8.2 until the 1st of January 2025

**Comment by tomglvng** [comment]
I close this PR: we only need to upgrade PHP to 8.2 (Symfony is still maintained until 1st of January 2025)
See my new PR --> https://github.com/akeneo/api-php-client/pull/290

**Pull request closed without merging**